### ISSUE TITLE
flag vs boolean query parameters?

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/chipper#519]
Should `cacheBuster` and `webgl` be of type 'boolean' instead of string?  They are both `true` by default, and their only practical use is with argument `false`, e.g. `webgl=false`).   Flags, on the other hand, are typically used without an argument.

**Comment by jonathanolson** [comment]
Or convert them to flags like 'noCacheBuster' and 'noWebGL'?

I don't have a strong preference, but they presumably should be at least boolean/default:true instead of string.

**Comment by pixelzoom** [comment]
Indeed, their schema would be: `{ type: 'boolean', defaultValue: true }`.

I'm not a fan of negative boolean/flag names (especially in code), but if it makes life easier...

**Comment by jonathanolson** [comment]
> Indeed, their schema would be: { type: 'boolean', defaultValue: true }.

That sounds good to me, the names just seem cleaner.

**Comment by pixelzoom** [comment]
I changed cacheBuster and webgl to type 'boolean'.

@samreid Looks like it would make more sense for these phet-io query parameters in SimIFrameClient to be type 'boolean':

```js
        options.emitStates = QueryStringMachine.get( 'phet-io.emitStates', {
          type: 'flag',
          defaultValue: options.emitStates
        } );
        options.emitInputEvents = QueryStringMachine.get( 'phet-io.emitInputEvents', {
          type: 'flag',
          defaultValue: options.emitInputEvents
        } );
```

Proposal:  Let's simplify the 'flag' type, so that it never takes a value and is simply an on/off switch -- on if present, off if absent.  This would cover 100% of our current uses of type 'flag' (with the exception of the 2 phet-io query parameters noted above).  It would also force us to use 'boolean' where appropriate, instead of `{ type: 'flag', defaultValue: true }`.

**Comment by samreid** [comment]
It has been confusing to have `defaultValue` for flag, so I also recommend to remove it.

**Comment by samreid** [comment]
On the other hand, it has been useful in debugging to be able to toggle ```http://phet.colorado.edu/sims/long-url/version/etc/sim.html?longQueryParameters&otherQueryParameters&webgl=false&otherQueryParameters```

to 

```http://phet.colorado.edu/sims/long-url/version/etc/sim.html?longQueryParameters&otherQueryParameters&webgl=true&otherQueryParameters```

And have it behave as expected.

But I'd still be happy with removing default value for flag type.  Let's discuss a bit further before I proceed with implementation changes.

**Comment by pixelzoom** [comment]
> Let's discuss a bit further before I proceed with implementation changes.

@jonathanolson Any feeling/thoughts on this?

**Comment by jonathanolson** [comment]
> @jonathanolson Any feeling/thoughts on this?

What is the proposal? Keeping ```?webgl=false``` seems cleaner than ```?disallowWebGL``` or an equivalent, so having defaults makes sense.

**Comment by pixelzoom** [comment]
Proposal in https://github.com/phetsims/chipper/issues/519#issuecomment-261613232:
> Proposal:  Let's simplify the 'flag' type, so that it never takes a value and is simply an on/off switch -- on if present, off if absent.  This would cover 100% of our current uses of type 'flag' (with the exception of the 2 phet-io query parameters noted above).  It would also force us to use 'boolean' where appropriate, instead of `{ type: 'flag', defaultValue: true }`.

**Comment by jonathanolson** [comment]
So 'on if present' would mean we'd need a flag like ```?disallowWebGL``` instead of ```?webgl=false```, correct?

**Comment by pixelzoom** [comment]
`webgl` is not a 'flag', it's type 'boolean'.

**Comment by jonathanolson** [comment]
Proposal for flags not having a default sounds great.

**Comment by pixelzoom** [comment]
Sounds like we're all in agreement that it would be good to removing defaultValue for 'flag' type.

**Comment by pixelzoom** [comment]
Wait on this until we discuss https://github.com/phetsims/query-string-machine/issues/19#issuecomment-262598021.

**Comment by samreid** [comment]
I converted flags with defaultValue to boolean in https://github.com/phetsims/chipper/commit/ccd9ef533cf4d25351141795ab57ce9014f4c430

I'll create a separate issue in query-string-machine that will help us enforce us that flags cannot have defaults.
